src/Device/Driver/LX_EOS/LXEosDownload.cpp

```diff
@@ -0,0 +1,245 @@
+  uint32_t bytes_remaining = flight.internal.lx_eos.file_size;
+
+  env.SetProgressRange(100);
+  env.SetProgressPosition(0);
+
+  // Download blocks until remaining bytes reach zero
+  for (uint16_t block_id = 0; bytes_remaining > 0; block_id++) {
+    AllocatedArray<std::byte> block;
+
+    // Try up to 5 times before giving up
+    for (uint8_t attempt = 0; attempt < 5; attempt++) {
+      try {
+        block = GetFlightLogBlock(flight_id, block_id, env);
+      } catch (...) {
+        success = false;
+        continue;
+      }
+      success = true;
+      break;
+    }
+
+    if (!success)
+      break;
+
+    bos.Write(block);
+
+    // Make sure that bytes remaining cannot underflow
+    if (block.size() > bytes_remaining)
+      return false;
+
+    bytes_remaining -= block.size();
+
+    // Progress in percents downloaded
+    float progress =
+      100.0f * (1.0f - static_cast<float>(bytes_remaining) /
+                         static_cast<float>(flight.internal.lx_eos.file_size));
+    env.SetProgressPosition(static_cast<unsigned int>(floor(progress)));
+  }
+
+  port.Flush();
+
+  if (success) {
+    bos.Flush();
+    fos.Commit();
+  }
+
+  env.SetProgressPosition(100);
+
+  port.StartRxThread();
+  return success;
+}
+
+bool
+LXEosDevice::GetNumberOfFlights(uint8_t& flight_count,
+                                OperationEnvironment& env)
+{
+  // Request
+  EosGetNumOfFlights data;
+  auto data_array = reinterpret_cast<std::byte*>(&data);
+  std::span<std::byte> message(data_array, sizeof(data));
+  if (!WriteAndWaitForACK(message, env))
+    return false;
+
+  // Read response (ACK byte already read)
+  EosNumOfFlightsResponse response;
+  try {
+    port.FullRead(
+      { reinterpret_cast<std::byte*>(&response) + 1, sizeof(response) - 1 },
+      env,
+      std::chrono::milliseconds(5000));
+  } catch (...) {
+    return false;
+  }
+
+  // Check CRC
+  if (Calculate8bitCRC(
+        reinterpret_cast<uint8_t*>(&response), sizeof(response), 0xFF, 0x69) !=
+      0x00)
+    return false;
+
+  flight_count = response.number_of_flights;
+  return true;
+}
+
+bool
+LXEosDevice::GetFlightInfo(const uint8_t index,
+                           RecordedFlightInfo& flight,
+                           OperationEnvironment& env)
+{
+  if (index == 0) // Invalid
+    return false;
+
+  // Request
+  EosRequestFlightInfo data;
+  data.flight_id = ToLE16(static_cast<uint16_t>(index));
+  data.crc = Calculate8bitCRC(
+    reinterpret_cast<uint8_t*>(&data), sizeof(data) - 1, 0xFF, 0x69);
+  std::span<std::byte> message(reinterpret_cast<std::byte*>(&data),
+                               sizeof(data));
+  if (!WriteAndWaitForACK(message, env))
+    return false;
+
+  // Read response (ACK byte already read)
+  EosFlightInfoResponse info;
+  try {
+    port.FullRead(
+      { reinterpret_cast<std::byte*>(&info) + 1, sizeof(info) - 1 },
+      env,
+      std::chrono::milliseconds(5000));
+  } catch (...) {
+    return false;
```

Comment:
Again: what's the point of this exception handler All it does is discard details about the error condition, but why?